### PR TITLE
Binary size reduction in KernelTypeStrResolver and GraphPartitioner

### PR DESCRIPTION
**Pull request by edgchen1**
### Description

Reduce binary size for minimal Android builds.
- reduce places where Status objects are created in KernelTypeStrResolver::LoadFromOrtFormat()
- remove some unused parameters (in a base minimal build) and code in graph_partitioner.cc

### Motivation and Context

Modest reduction of binary size (-2.8K) in minimal Android build.

bloaty output, comparing 42c0a21901 (updated) with 5c89c37f7f (baseline):
```
     VM SIZE
 --------------
  +1.8%     +60    D:/source/onnxruntime/onnxruntime/core/framework/graph_partitioner.cc
    [NEW]    +256    onnxruntime::KernelLookup::KernelLookup()
    [NEW]    +244    onnxruntime::KernelRegistryManager::GetKernelRegistriesByProviderType()
    [NEW]     +80    onnxruntime::MakeString<>()
   -38.6%    -520    onnxruntime::PartitionOrtFormatModelImpl()
  +4.0%     +24    [LOAD #1 [RX]]
  -0.2%    -240    [section .rodata]
  -0.2%    -692    [section .text]
    +3.9%    +792    onnxruntime::detail::MakeStringImpl<>()
    [DEL]      -8    __CortexA53843419_103004
    -2.2%     -80    onnxruntime::MakeString<>()
    [DEL]    -244    onnxruntime::KernelRegistryManager::GetKernelRegistriesByProviderType()
    [DEL]    -256    onnxruntime::KernelLookup::KernelLookup()
   -32.7%    -896    std::__ndk1::__hash_table<>::__rehash()
 -21.3% -2.02Ki    D:/source/onnxruntime/onnxruntime/core/framework/kernel_type_str_resolver.cc
    [NEW]    +896    std::__ndk1::__hash_table<>::__rehash()
   -43.9% -1.06Ki    onnxruntime::KernelTypeStrResolver::LoadFromOrtFormat()
   -68.4% -1.83Ki    onnxruntime::detail::MakeStringImpl<>()
  -0.2% -2.84Ki    TOTAL
```

Note: the graph_partitioner.cc compileunit appears to increase in size, but some items seem to have been moved there from `[section .text]`.

### CODE REVIEW

File: onnxruntime/core/framework/graph_partitioner.cc [hunk 1]
```diff
@@ -570,16 +593,27 @@ static Status PartitionOrtFormatModelImpl(const PartitionParams& partition_param
...
-                                         GraphPartitioner::Mode::kOrtFormatLoad, capabilities,
-                                         transform_layout_function));
+  const auto get_capability_params = GetCapabilityForEPParams{
+      std::ref(graph),
+      std::cref(kernel_registry_mgr),
+      std::ref(current_ep),
+      std::ref(capabilities),
+      GraphPartitioner::Mode::kOrtFormatLoad,
+      partition_params.transform_layout_function,
+  };
```
Comment (skottmckay) [inline onnxruntime/core/framework/graph_partitioner.cc:606]: Can we put the ifdef around these two params to reduce duplication?

**Comment by cloudhan** [comment]
Off topic, when working on #11071, I notice that using `ostringstream` introduce many locale related object, and contribute a huge portion of binary size. If you are able to remove `std::basic_ostream` usage in min built, it might improve binary size by ~80kb

File: onnxruntime/core/framework/graph_partitioner.cc (hunk 1 above)
Comment (edgchen1) [reply onnxruntime/core/framework/graph_partitioner.cc]: clang-format didn't like the ifdef there so I disabled it for those lines

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**